wasp/apps/musicplayer.py

```diff
@@ -0,0 +1,208 @@
+                self._state_changed = True
+        else:
+            self._state_changed = False
+        wasp.system.musicstate = {}
+        if music_track_now:
+            if music_track_now != self._track:
+                self._track = music_track_now
+                self._track_changed = True
+        else:
+            self._track_changed = False
+        if music_artist_now:
+            if music_artist_now != self._artist:
+                self._artist = music_artist_now
+                self._artist_changed = True
+        else:
+            self._artist_changed = False
+        wasp.system.musicinfo = {}
+        self._update()
+
+    def swipe(self, event):
+        """
+        Notify the application of a touchscreen swipe event.
+        """
+        if event[0] == wasp.EventType.UP:
+            self._send_cmd(self._gb_vup)
+        elif event[0] == wasp.EventType.DOWN:
+            self._send_cmd(self._gb_vdo)
+        elif event[0] == wasp.EventType.LEFT:
+            self._send_cmd(self._gb_next)
+        elif event[0] == wasp.EventType.RIGHT:
+            self._send_cmd(self._gb_prev)
+
+    def touch(self, event):
+        self._play_state = not self._play_state
+        if self._play_state:
+            self._musicstate = 'play'
+            self._icon_state = self._pause_icon
+            self._draw_button()
+            self._send_cmd(self._gb_play)
+        else:
+            self._musicstate = 'pause'
+            self._icon_state = self._play_icon
+            self._draw_button()
+            self._send_cmd(self._gb_pause)
+
+    def draw(self):
+        """Redraw the display from scratch."""
+        self._draw()
+
+    def _draw(self):
+        """Redraw the updated zones."""
+        if self._state_changed:
+            self._draw_button()
+        if self._track_changed:
+            self._draw_track()
+        if self._artist_changed:
+            self._draw_artist()
+
+    def _draw_button(self):
+        """Redraw player button"""
+        self._fill_space('mid')
+        wasp.watch.drawable.blit(self._icon_state, self._center_at,
+                                 self._center_at)
+
+    def _draw_artist(self):
+        """Redraw artist info"""
+        if self._artist:
+            draw = wasp.watch.drawable
+            chunks = draw.wrap(self._artist, 240)
+            self._fill_space('top')
+            for i in range(len(chunks)-1):
+                sub = self._artist[chunks[i]:chunks[i+1]].rstrip()
+                draw.string(sub, 0, self._artist_pos + 24 * i, 240)
+
+    def _draw_track(self):
+        """Redraw track info"""
+        if self._track:
+            draw = wasp.watch.drawable
+            chunks = draw.wrap(self._track, 240)
+            self._fill_space('down')
+            for i in range(len(chunks)-1):
+                sub = self._track[chunks[i]:chunks[i+1]].rstrip()
+                draw.string(sub, 0, self._track_pos + 24 * i, 240)
```

Comment:
These two functions seem "wrong"

They basically have identical structure and are only parameterized by the variable they act on (self._track in this case) and something with a similar name (self._track_pos). Can't we have_draw_label(self, label, pos)?